packages/esm-service-queues-app/src/index.ts

```diff
@@ -95,9 +94,33 @@ export const addProviderToRoomModal = getAsyncLifecycle(
 
 export const addQueueEntry = getSyncLifecycle(addQueueEntryComponent, options);
 
-export const queueTableName = getSyncLifecycle(QueueTableName, options);
-export const queueTablePriority = getSyncLifecycle(QueueTablePriority, options);
-export const queueTableStatus = getSyncLifecycle(QueueTableStatus, options);
+export const queueTableName = getAsyncLifecycle(
+  () => import('./queue-table/queue-table-cells.component').then(({ QueueTableName }) => ({ default: QueueTableName })),
+  {
+    featureName: 'queue table name column',
+    moduleName,
+  },
+);
+
+export const queueTablePriority = getAsyncLifecycle(
+  () =>
+    import('./queue-table/queue-table-cells.component').then(({ QueueTablePriority }) => ({
+      default: QueueTablePriority,
+    })),
+  {
+    featureName: 'queue table name column',
+    moduleName,
+  },
+);
+
+export const queueTableStatus = getAsyncLifecycle(
+  () =>
+    import('./queue-table/queue-table-cells.component').then(({ QueueTableStatus }) => ({ default: QueueTableStatus })),
+  {
+    featureName: 'queue table name column',
+    moduleName,
+  },
+);
```

Comment:
Can you explain why this has been modified like this, instead of just changing "getSyncLifecycle" to "get AsyncLifecycle"?

Reply:
The two functions don't take in the same inputs. `getAsyncLifecycle` expects a `Promise` returned by dynamically calling `import("***.component")`, where `***.component.tsx` is usually a file that defines a Component that is exported with `default export`'. In this case, `queue-table-cells.component.tsx` has several exports, and therefore no `default` export, This syntax is admittedly uglier. The alternative would be to create separate files for each of those 3 components defined in `queue-table-cells.component.tsx`.

Reply:
OK, defer to others like @ibacher on this.  I don't feel like I tend to see it like this in the code elsewhere, so that might imply that the right thing is to move the components into separate files with default exports?

Reply:
Yeah, generally, the component should be the default export, even if other things are exported (although that sounds suspiciously like code that shouldn't all be in one component file; usually a component file should just define a single React component and if things need to be shared, they go elsewhere).

Reply:
Yeah, I need to refresh my memory on how all this stuff works, more or less defer to @ibacher  on this one